app/grandchallenge/reader_studies/tasks.py

```diff
@@ -54,6 +54,35 @@ def add_scores_for_display_set(*, instance_pk, ds_pk):
         add_score(instance, ground_truth.answer)
 
 
+@acks_late_2xlarge_task
+def bulk_assign_scores_for_reader_study(*, reader_study_pk):
+    ground_truth = Answer.objects.filter(
+        question__reader_study__pk=reader_study_pk,
+        is_ground_truth=True,
+    ).all()
+
+    def key(answer):
+        return (str(answer.question_id), str(answer.display_set_id))
+
+    ground_truth_lookup = {}
+    for answer in ground_truth:
+        ground_truth_lookup[key(answer)] = answer
+
+    answers = Answer.objects.filter(
+        question__reader_study__pk=reader_study_pk
+    ).prefetch_related("question")
```

Comment:
I think only non-ground-truth answers should be set:

```suggestion
    answers = Answer.objects.filter(
        question__reader_study__pk=reader_study_pk,
        is_ground_truth=False,
    ).prefetch_related("question")
```

Reply:
Agreed! Albeit there are filters that make this redundant I still think we should have a clear split here.